model_processing/crates/webify_models/src/mesh_update/rename_image_references.rs

```diff
@@ -0,0 +1,173 @@
+//! Rename all image references inside a DAE mesh to be .PNG and point
+//! at the right textures path
+
+use std::path::PathBuf;
+use std::{
+    fs::{self},
+    path::Path,
+};
+
+use aho_corasick::AhoCorasickBuilder;
+
+/// Orchestrator to rename image references in a DAE mesh
+pub fn rename_image_references(mesh: &PathBuf) -> std::result::Result<(), std::io::Error> {
+    let result = find_and_rename_image_references(mesh)?;
+    let final_result = update_texture_path(result)?;
+    fs::write(mesh, final_result)?;
+
+    Ok(())
+}
+
+/// Rename all occurences of supported image types to PNG
+fn find_and_rename_image_references(mesh: &PathBuf) -> std::result::Result<String, std::io::Error> {
+    let patterns = &[".tga", "_tga", ".jpg", "_jpg", ".jpeg", "_jpeg", ".gif", "_gif"];
+    let f = fs::read_to_string(mesh)?;
+
+    let ac = AhoCorasickBuilder::new().build(patterns);
+    let result = ac.replace_all(&f, &[".png", "_png", ".png", "_png", ".png", "_png", ".png", "_png"]);
+
+    Ok(result)
+}
+
+/// Update all texture paths to point to the relative path for textures/materials
+fn update_texture_path(contents: String) -> std::result::Result<String, std::io::Error> {
+    let texture_element = &["<init_from>", "</init_from>"];
+    let ac = AhoCorasickBuilder::new().build(texture_element);
+    let texture_path = Path::new("materials").join("textures");
+
+    let mut new_contents = String::new();
+
+    // Find all texture elements
+    for line in contents.lines() {
```

Comment:
I think a lot of this could refactored using more functional iterators such as `fold` and `map`. I'm willing to look more into if that's something you're interested in doing, or maybe it's better left for another time.

Reply:
Maybe for another time for now, but thanks!